static/app/utils/buttonClick/useButtonClick.tsx

```diff
@@ -0,0 +1,77 @@
+import {useCallback, useContext} from 'react';
+
+import {ButtonProps} from 'sentry/components/button';
+import HookStore from 'sentry/stores/hookStore';
+import {OrganizationContext} from 'sentry/views/organizationContext';
+import {RouteContext} from 'sentry/views/routeContext';
+
+type Props = ButtonProps;
+
+export const trackButtonClick = HookStore.get('analytics:track-button-clicks')[0];
+
+export default function useButtonClick({
```

Comment:
@roggenkemper why not derive `useButtonClick` from getsentry the same way we do for `useRouteActivatedHook`? We should try to replicate the same sort of architecture. This code doesn't really need to be in Sentry I think.